### PR TITLE
Javascript ES6 preprocessor with Babel

### PR DESCRIPTION
**Pull request by Kauabunga**
Added a follow up question to the first language selection question to enable ES6 client support with Babel if they originally select 'Javascript'.

Using the Babel library after better experiences with it over Traceur. Has more readable output and does not require a runtime library component.

* Change scoped to the app generator
* Added generator tests
* Does not touch the generated ES5 angular components
* Does not touch the language used in the server components

*Related issue: Any plans for es6 and traceur support? #684*

### CODE REVIEW

**Comment by Awk34** [comment]
:+1: Nice

**Comment by bjackson** [comment]
This would be really sweet. It looks like there are some merge conflicts that might need to be solved before it can be merged, though.

File: app/templates/client/index.html [hunk 1]
```diff
@@ -50,7 +50,11 @@
       <script src="socket.io-client/socket.io.js"></script><% } %>
     <!-- endbuild -->
 
+        <% if(filters.babel) { %>
+        <!-- build:js(.tmp) app/app.js -->
```
Comment (bjackson) [inline app/templates/client/index.html]: What does this do exactly? Why is it different depending on wether babel is used?

**Comment by Awk34** [comment]
Also just thought, can you add the babel karma preprocessor and make sure that tests run alright?

File: app/templates/client/index.html (hunk 1 above)
Comment (Kauabunga) [inline app/templates/client/index.html]: Heya - bit of a slow reply! Babel is precompiling all the /client js resources to /.tmp. 

If we inject all the js resources from /.tmp and /client with the non-babel index.html annotation  ....   ..... then we get a double up of all ES5 and ES6 items causing angular to die with multiple definitions of modules or something along those lines.

**Comment by Kauabunga** [comment]
Hey Andrew - have just booted up this fork again. Will have a look at getting babel and karma dancing when my time frees up this week.

**Comment by Kauabunga** [comment]
Also - re the merge conflicts. I had forked from master though submitted the pull request to canary as suggested in the readme. Should I have (forked from canary/merged from canary) before submitting the request?

**Comment by Awk34** [comment]
As Canary has become a bit stale, it's fine to merge to master, especially with a fully working feature. If you don't have the time to tweak this I can go for it.

**Comment by Kauabunga** [comment]
Sure - Happy for you to gun for it. 

I had been playing around with karma config to get ES6 client tests working and have just wrapped up getting the changes into the generator.

Check it out if you'd like - they're sitting on Kauabunga:pullrequest-feature-es6-babel-and-karma

Seems to merge with DaftMonk:master all good.

**Comment by Awk34** [comment]
will do, thanks!

**Comment by Awk34** [comment]
Closed with #998, thank you!

**Pull request closed without merging**